projects/plugins/social/src/js/components/admin-page/index.jsx

```diff
@@ -28,6 +28,18 @@ const Admin = () => {
 	const showConnectionCard = ! isRegistered || ! isUserConnected;
 	const [ forceDisplayPricingPage, setForceDisplayPricingPage ] = useState( false );
 
+	const refreshJetpackSocialSettings = useDispatch( SOCIAL_STORE_ID ).refreshJetpackSocialSettings;
+
+	useEffect( () => {
+		if (
+			isModuleEnabled &&
+			isUpdatingJetpackSettings &&
```

Comment:
I think it makes sense to update the store while the main module is turning on, this way it will be refreshed already when the toggles appear, reducing the page jank